### PR TITLE
Handle the case where the voName is shorter than 3 characters.

### LINKED ISSUE #3: VOConfigTask fails when the VO name is too short
When the name of the VO is too short (less than 3 characters), the VOConfigTask fails with the following error:

```
quattor/quattor.build.xml:434: java.lang.StringIndexOutOfBoundsException: String index out of range: 3
at java.lang.String.substring(String.java:1963)
at org.quattor.ant.VOConfigTask$VOConfig.setAccountPrefix(VOConfigTask.java:641)
at org.quattor.ant.VOConfigTask$VOConfig.getAccountPrefix(VOConfigTask.java:601)
at org.quattor.ant.VOConfigTask$VOConfig.writeVOTemplate(VOConfigTask.java:700)
at org.quattor.ant.VOConfigTask$VOConfig.access$000(VOConfigTask.java:562)
at org.quattor.ant.VOConfigTask.execute(VOConfigTask.java:277)
at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:497)
at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
at org.apache.tools.ant.Task.perform(Task.java:348)
at org.apache.tools.ant.Target.execute(Target.java:435)
at org.apache.tools.ant.Target.performTasks(Target.java:456)
at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
at org.apache.tools.ant.Main.runBuild(Main.java:851)
at org.apache.tools.ant.Main.startAnt(Main.java:235)
at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
```

### PR DESCRIPTION
**Pull request by Pansanel**
Fix #3

### CODE REVIEW

File: src/main/java/VOConfigTask.java [hunk 1]
```diff
@@ -647,9 +647,16 @@ public String getPilotRoleFQAN() {
...
+            }
+            String voName = getName().replaceFirst("^vo\\.", "").replaceAll("[^A-Za-z0-9]", "");
+            if ( voName.length() >= 3 ) {
+                this.accountPrefix = voName.substring(0,3);
+            } else if ( voName.length() == 2 ) {
+                this.accountPrefix = voName.substring(0,2);
+            } else {
+                throw new BuildException("The VO name is too short: cannot generate account prefix");
             }
-            this.accountPrefix = getName().replaceFirst("^vo\\.", "").replaceAll("[^A-Za-z0-9]", "").substring(0,3);
```
Comment (jouvin) [inline src/main/java/VOConfigTask.java:652]: I am not sure to understand the intent of this removal. These replace are important to normalize the account prefix. If `substring(0,3)` is the problem when the account prefix is shorter than 3 chars, it is probably enough to replace `3` by the minimum between `3` and the length of the VO name after the first replace. Did I miss something?

Comment (Pansanel) [inline src/main/java/VOConfigTask.java]: If you look at the code, when the length of the voName is >= 3, you get exactly
String voName = getName().replaceFirst("^vo\\.", "").replaceAll("[^A-Za-z0-9]", "");
this.accountPrefix = voName.substring(0,3);
which is same as:
this.accountPrefix = getName().replaceFirst("^vo\\.", "").replaceAll("[^A-Za-z0-9]", "").substring(0,3);

**Comment by Pansanel** [comment]
The fix has been tested successfully at IPHC. The jar file has been sent to M3PEC for further testing.

**Comment by Pansanel** [comment]
This PR has been accidentally closed. It has been reopened as #11

**Pull request closed without merging**